app/scripts/data-analytics/directives/data-entry-stats-view-four.directive.js

```diff
@@ -86,7 +89,21 @@ jshint -W003, -W026
 
 		activate();
 		function activate() {
-			//loadStatsFromServer();
+      $scope.viewCachedData = helperService.getSetViewCachedData();
+      if($scope.viewCachedData.cached && $scope.viewCachedData.viewId==='view4') {
+        $timeout(function(){
+          $scope.selectedLocations = $scope.viewCachedData.selectedLocations;
+          $scope.selectedProvider = $scope.viewCachedData.selectedProvider;
+      		$scope.selectedEncounterTypes = $scope.viewCachedData.selectedEncounterTypes;
+      		$scope.selectedForms = $scope.viewCachedData.selectedForms;
+      		$scope.startDate = $scope.viewCachedData.startDate;
+      		$scope.endDate = $scope.viewCachedData.endDate
+          $scope.selectedCreator = $scope.viewCachedData.selectedCreator
+          console.log('cached called', $scope.viewCachedData);
+          loadStatsFromServer();
+        }, 5000);
```

Comment:
We need to figure out how to not rely on this timeout as it will fail if fetching of locations takes more than 5000 miliseconds. I think the locations are cached, so if you try with a lesser value, say 100 miliseconds, I think it should work